### PR TITLE
terminal: Evolution on input printing and flushing

### PR DESCRIPTION
**Pull request by doniacld**
- Add carriage return
- Add methods for writing and flushing input

I have been working with @gdey and it would be great if he can review it.

### CODE REVIEW

**Review by gdey — APPROVED** [review]
LGTM

There is a minor change though, we no longer need two function fro flushing the input to the serial and screen, so it would make sense to combine `writeInput` and  `flushInput`

**Comment by doniacld** [comment]
> LGTM
> 
> There is a minor change though, we no longer need two function fro flushing the input to the serial and screen, so it would make sense to combine `writeInput` and `flushInput`

I just moved the write and flush in one method and added some comments. Let me know what you think after these las changes.

File: tutorial/terminal/main.go [hunk 1]
```diff
@@ -49,36 +62,54 @@ func main() {
...
+				writeInputAndFlush(input, i)
 				i = 0
+			// any other character, let's echo it
 			default:
 				// just echo the character
 				uart.WriteByte(data)
 				input[i] = data
 
-				terminal.WriteByte(data)
+				_ = terminal.WriteByte(data)
```
Comment (deadprogram) [inline tutorial/terminal/main.go:82]: Why all the `_ =`?

Comment (doniacld) [reply tutorial/terminal/main.go:82]: My linter was unhappy with uncaught errors 🤷‍♀️ I can revert them if you think it's more readable.

Comment (deadprogram) [reply tutorial/terminal/main.go:82]: it is very unreadable with all the `_` IMO :crying_cat_face:

Comment (doniacld) [reply tutorial/terminal/main.go:82]: Fixed! :)

**Comment by doniacld** [comment]
Can I add the label `hacktoberfest-accepted` ? 🥺

**Comment by deadprogram** [comment]
Thanks @doniacld now merging!

**Pull request merged**